// sheetdb-node/index.js
// sheetdb = require(sheetdb-node);
// (ES6+) import sheetdb from 'sheetdb-node';
//
// sheetdb.create(newRow, sheet);
// sheetdb.read(limit, offset, search, sheet);
// sheetdb.update(columnName, value, newRow, updateWhole, sheet);
// sheetdb.delete(columnName, value, sheet);

import readFunc from './lib/read.js';
import createFunc from './lib/create.js';
import updateFunc from './lib/update.js';
import deleteFunc from './lib/delete.js';
import endpointFunc from './lib/endpoint.js';
import validAddress from './lib/validAddress.js';
import isURL from './lib/isURL.js';


var sheetdbNode = function(config) {
  var configParam = config || {};

  configParam.version = configParam.version || '1';
  configParam.auth_login = configParam.auth_login || '';
  configParam.auth_password = configParam.auth_password || '';

  if(!configParam.address) {
    throw Error('address param needed');
  }

  if(!validAddress(configParam.address)) {
    throw Error('wrong address param.');
  }

  if(!isURL(configParam.address)) {
    configParam.address = 'https://sheetdb.io/api/v' +
      configParam.version + '/' +
      configParam.address;
  }

  var address = configParam.address;

  return {
    config: configParam,
    create: createFunc,
    read: readFunc,
    update: updateFunc,
    delete: deleteFunc,
    endpoint: endpointFunc,
  }
}

export default sheetdbNode;
